data/profile-physlr.Rmd

```diff
@@ -0,0 +1,384 @@
+  {
+    df[i, "time_min"] = 1
+  }
+  df[i, "total_time_min"] = df[i, "time_hr"] * 60 + df[i, "time_min"]
+
+  ## Memory
+  mem_line_no = grep('Maximum', lines_of_file)
+  mem_line <- lines_of_file[mem_line_no]
+  mem_loci <- str_locate_all(mem_line, "\\:")
+  len = nchar(mem_line)
+  df[i, "mem"] = strtoi(substr(mem_line, mem_loci[[1]][1] + 1 , len))
+
+  ## Name and programming language
+  py_pattern = str_match(lines_of_file[1], regex('/bin/physlr\ \\S*'))
+  if (!is.na(py_pattern[1, 1]))
+  {
+    #name_loci = str_locate_all(lines_of_file[1], "bin/physlr")
+    df[i, "step"] = str_match(py_pattern,
+                              regex('(?<=\ ).*',
+                                    ignore.case = TRUE, perl =
+                                      TRUE))
+    df[i, "pl"] = "python"
+  } else
+  {
+    cpp_pattern = str_match(lines_of_file[1], regex('/src/physlr-\\S*'))
+    if (!is.na(cpp_pattern[1, 1]))
+    {
+      df[i, "step"] = str_match(cpp_pattern,
+                                regex('(?<=-).*',
+                                      ignore.case = TRUE, perl =
+                                        TRUE))
+      df[i, "pl"] = "c/c++"
+    } else
+    {
+      pattern = str_match(lines_of_file[1], regex('being timed:\ \"\\S*'))
+      if (!is.na(pattern[1, 1]))
+      {
+        df[i, "step"] = str_match(pattern,
+                                  regex(
+                                    '(?<=being timed:\ \").*',
+                                    ignore.case = TRUE,
+                                    perl = TRUE
+                                  ))
+        df[i, "pl"] = "third-party"
+      } else
+      {
+        cat("Unknown command patrtern:\n")
+        print(lines_of_file[1])
+        df[i, "step"] = "unknown name"
+        df[i, "pl"] = "unknown"
+      }
+    }
+  }
+}
+df_temp = df
+df = df_temp
+if (dim(df)[1] == 0) {
+  cat("There is no correct *.time file to extract results from!")
+  good = FALSE
+} else if (sum(is.na(df[, 1])) == dim(df)[1]) {
+  cat("No correct *.time file to extract results from!")
+  good = FALSE
+} else{
+
+  to_be_removed = which(is.na(df[, 1]))
+
+  if (length(to_be_removed) > 0) {
+    cat("Following file(s) will be skipped due to having no step name\n")
+    print(sorted_time_files[to_be_removed])
+  }
+
+  df = df[!is.na(df[, 1]),]
+
+  if (dim(df)[1] == 0) {
+    cat("No correct *.time file to extract results from!")
+    good = FALSE
+  } else{
+    to_be_removed2 = array()
+
+    for (i in 1:dim(df)[1]) {
+      if (sum(is.na(df[i,])) > 0) {
+        to_be_removed2[length(to_be_removed2)[1] + 1] = i
+      }
+    }
+
+    if (length(to_be_removed2) > 1) {
+      to_be_removed2 = to_be_removed2[2:length(to_be_removed2)]
+      cat("Following file(s) will be excluded due to having NA values")
+      print(print(sorted_time_files[to_be_removed2]))
+      print(df[to_be_removed2,])
+      df = df[!to_be_removed2,]
+    }
+
+    if (dim(df)[1] == 0) {
```

Comment:
Could we just check this at the end? Seems like it is repeated a lot

Reply:
I keep changing the `df` dataframe, so these if-statements are checking the file for different issues. I believe that some cases would not happen, but I just wanted to take into account any possibility...

Reply:
Hmm..But if it that statement was true at any point in this code, wouldn't it still be true in the end? And you seem to have the same error message for each of them?

Reply:
Umm, yes I can change the message to be more specific about the problem but that's not the main point... The main reason to check it multiple times is that if the statement is true the first time, then the operations I accomplish in the `else` part are not valid and running them results in error... I mean I am avoiding run-time errors this way...

Reply:
My point wasn't about the error messages needing to be different (I don't think that is necessary), but about the repeated code which isn't great practice. You know best all the conditions you are checking for and how realistic the situations are (I find this part of the code a bit hard to follow), so I suggest looking at possibilities for refactoring to make it more concise.

Reply:
Sorry I was not good at answering this...
I just reformatted the code and now it checks once for all issues and then tries to remove rows with `NA` values.